drivers/sensors/l86xxx_uorb.c

```diff
@@ -0,0 +1,754 @@
+{
+  FAR l86xxx_dev_s *dev = container_of(lower, FAR l86xxx_dev_s, lower);
+  return send_command(dev, (L86XXX_PMTK_COMMAND)cmd, arg);
+}
+
+/****************************************************************************
+ * Name: nau7802_activate
+ ****************************************************************************/
+
+static int l86xxx_activate(FAR struct sensor_lowerhalf_s *lower,
+                              FAR struct file *filep, bool enable)
+{
+  FAR l86xxx_dev_s *dev = container_of(lower, FAR l86xxx_dev_s, lower);
+
+  /* If not already enabled, start gps */
+
+  if (enable && !dev->enabled)
+    {
+      nxsem_post(&dev->run);
+      dev->enabled = true;
+      send_command(dev, CMD_HOT_START, (int)NULL);
+    }
+
+  /* If not already disabled, send gps into standby mode */
+
+  else if (!enable && dev->enabled)
+    {
+      dev->enabled = false;
+      send_command(dev, CMD_STANDBY_MODE, 0);
+    }
+
+  return 0;
+}
+
+/****************************************************************************
+ * Name: l86xxx_set_interval
+ *
+ * Description:
+ *   Set position fix interval of L86-XXX GNSS module
+ *
+ * Returns:
+ *   -1 if invalid interval, else return value from send_command
+ ****************************************************************************/
+
+static int l86xxx_set_interval(FAR struct sensor_lowerhalf_s *lower,
+                                    FAR struct file *filep,
+                                    FAR uint32_t *period_us)
+{
+  FAR l86xxx_dev_s *dev = container_of(lower, FAR l86xxx_dev_s, lower);
+  int fix_interval = *period_us;
+  if (fix_interval < 100 || fix_interval > 10000)
+    {
+      return -1;
+    }
+
+  int ret = send_command(dev, SET_POS_FIX, fix_interval);
+  return ret;
+}
+
+/****************************************************************************
+ * Name: l86xxx_thread
+ *
+ * Description:
+ *   Kernel thread to poll the l86xxx
+ ****************************************************************************/
+
+static int l86xxx_thread(int argc, FAR char *argv[])
+{
+  FAR l86xxx_dev_s *dev =
+      (FAR l86xxx_dev_s *)((uintptr_t)strtoul(argv[1], NULL, 16));
+  struct sensor_gnss gps;
+  memset(&gps, 0, sizeof(gps));
+  dev->enabled = true;
+  int err;
+
+  /* Read full line of NMEA output */
+
+  for (; ; )
+    {
+      /* If the sensor is disabled
+       * wait until enabled with activate function
+       */
+
+      if (!dev->enabled)
+        {
+          err = nxsem_wait(&dev->run);
+          if (err < 0)
+            {
+              continue;
+            }
+        }
+
+      /* Mutex required because some commands send ACKS */
+
+      nxmutex_lock(&dev->devlock);
+      read_line(dev);
+
+      /* Parse line based on NMEA sentence type */
+
+      switch (minmea_sentence_id(dev->buffer, false))
+      {
+        /* Time data is obtained from RMC sentence */
+
+        case MINMEA_SENTENCE_RMC:
```

Comment:
the similar code already done in gnss upperhalf, you can reuse the common code to simplify this driver